tests/cpp/test_matmul_scheduler.cpp

```diff
@@ -2736,6 +2737,76 @@ TEST_P(MatmulFusionTest, Llama2FFN) {
   }
 }
 
+// Test that we can schedule a Fusion containing two matmuls
+TEST_F(MatmulSchedulerTest, Llama2FFNNoSegmentation) {
+  NVFUSER_TEST_CUDA_ARCH_RANGE_GUARD(7, 5, 9, 0);
+
+  EnableOptionsGuard option_guard;
+  EnableOptionsGuard::getCurOptions().set(EnableOption::FuseMultipleMatmuls);
+
+  auto fusion_ptr = std::make_unique<Fusion>();
+  Fusion* fusion = fusion_ptr.get();
+  FusionGuard fg(fusion);
+
+  auto tv0 = makeContigTensor(2, DataType::Half);
+  auto tv1 = makeContigTensor(2, DataType::Half);
+  auto tv2 = makeContigTensor(2, DataType::Half);
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addInput(tv2);
+
+  auto tv3 = matmul(tv0, tv1);
+  auto tv4 = matmul(tv0, tv2);
+
+  // silu
+  auto tv5 = mul(sigmoid(tv3), tv3);
+
+  auto tv6 = mul(tv5, tv4);
+
+  fusion->addOutput(tv6);
+
+  const int M = 504, N = 136, K = 248;
+
+  at::manual_seed(0);
+  auto options = at::TensorOptions().dtype(at::kHalf).device(at::kCUDA);
+  auto t0 = at::randn({M, K}, options);
+  auto t1 = at::randn({K, N}, options);
+  auto t2 = at::randn({K, N}, options);
+  std::vector<c10::IValue> inputs{t0, t1, t2};
+
+  // These params are meant to exercise most of the available options
+  MatmulParams params;
+  params.supported_vec_size = {8, 8, 4};
+  params.mma_macro = MmaMacro::Ampere_16_8_16;
+  MatMulTileOptions gemm_tile;
+  gemm_tile.cta_tile = GemmTile(128, 64, 32);
+  gemm_tile.warp_tile = GemmTile(64, 64, 32);
+  gemm_tile.instruction_tile = GemmTile(16, 8, 16);
+  params.tile_sizes = gemm_tile;
+  params.cta_order = MatmulParams::TileRasterizationOrder::ColumnMajor;
+  params.grid_swizzle_factor = 2;
+  params.async_gmem_load_operands = true;
+  params.circular_buffer_options.circular_buffer_smem_write = true;
+  params.circular_buffer_options.circular_buffer_smem_read = true;
+  params.circular_buffer_options.smem_circular_buffer_stage = 2;
+  // params.splitk_factor = 2;
+  // params.use_smem_epilogue = true;
+  // params.promote_prologue_smem_reuse = true;
+  scheduleMultipleMatmuls(fusion, params);
+
+  FusionExecutor fe;
+  fe.compileFusion(fusion, inputs);
+
+  auto outputs = fe.runFusion(inputs);
+
+  auto t3 = at::matmul(t0.to(at::kFloat), t1.to(at::kFloat));
+  auto t4 = at::matmul(t0.to(at::kFloat), t2.to(at::kFloat));
+  auto t5 = at::sigmoid(t3) * t3;
+  auto tref = t5 * t4;
+
+  NVF_CHECK(outputs[0].allclose(tref, 0.005, 0.005));
```

Comment:
This test generates a kernel that computes two matmuls in a single kernel. It is currently passing (note the relaxed tolerance) with the following generated kernel:
```c++
__global__ void nvfuser_none_f0_c0_r0_g0(Tensor<__half, 2, 2> T0, Tensor<__half, 2, 2> T1, Tensor<__half, 2, 2> T2, Tensor<float, 2, 2> T10) {
  alignas(16) extern __shared__ char array[];
  const unsigned smem_offset = 0;
  NVFUSER_DEFINE_MAGIC_ZERO;
  nvfuser_index_t i0;
  i0 = ceilDiv(T0.logical_size[1LL], 32LL);
  nvfuser_index_t i1;
  i1 = 4LL * T1.logical_size[1LL];
  nvfuser_index_t i2;
  i2 = ((nvfuser_index_t)threadIdx.x) / 8LL;
  nvfuser_index_t i3;
  i3 = ((nvfuser_index_t)threadIdx.x) % 8LL;
  nvfuser_index_t i4;
  i4 = 8LL * i3;
  nvfuser_index_t i5;
  i5 = T1.logical_size[1LL] * i2;
  nvfuser_index_t i6;
  i6 = i1 * ((nvfuser_index_t)threadIdx.y);
  nvfuser_index_t i7;
  i7 = i1 * ((nvfuser_index_t)threadIdx.z);
  nvfuser_index_t i8;
  i8 = 64LL * (((nvfuser_index_t)blockIdx.x) / 2LL);
  __half* ptr9;
  ptr9 = ((((T2.data + i4) + i5) + i6) + i7) + i8;
  nvfuser_index_t i10;
  i10 = 8LL * T1.logical_size[1LL];
  nvfuser_index_t i11;
  i11 = 128LL * i2;
  nvfuser_index_t i12;
  i12 = 128LL * (i3 / 8LL);
  nvfuser_index_t i13;
  i13 = 512LL * ((nvfuser_index_t)threadIdx.y);
  nvfuser_index_t i14;
  i14 = 512LL * ((nvfuser_index_t)threadIdx.z);
  nvfuser_index_t i15;
  i15 = 4LL * ((nvfuser_index_t)threadIdx.y);
  nvfuser_index_t i16;
  i16 = 4LL * ((nvfuser_index_t)threadIdx.z);
  nvfuser_index_t i17;
  i17 = 16LL * ((i3 % 8LL) ^ (((i2 + i15) + i16) % 8LL));
  __half* T13 = reinterpret_cast<__half*>(array + smem_offset + 24576LL);
  unsigned i18;
  i18 = toSmem(T13);
  unsigned i19;
  i19 = ((((i18 + i11) + i12) + i13) + i14) + i17;
  __half* ptr20;
  ptr20 = ((((T1.data + i4) + i5) + i6) + i7) + i8;
  __half* T12 = reinterpret_cast<__half*>(array + smem_offset + 16384LL);
  unsigned i21;
  i21 = toSmem(T12);
  unsigned i22;
  i22 = ((((i21 + i11) + i12) + i13) + i14) + i17;
  nvfuser_index_t i23;
  i23 = 8LL * T0.logical_size[1LL];
  nvfuser_index_t i24;
  i24 = ((nvfuser_index_t)threadIdx.x) / 4LL;
  nvfuser_index_t i25;
  i25 = ((nvfuser_index_t)threadIdx.x) % 4LL;
  nvfuser_index_t i26;
  i26 = 8LL * i25;
  nvfuser_index_t i27;
  i27 = T0.logical_size[1LL] * i24;
  nvfuser_index_t i28;
  i28 = i23 * ((nvfuser_index_t)threadIdx.y);
  nvfuser_index_t i29;
  i29 = i23 * ((nvfuser_index_t)threadIdx.z);
  nvfuser_index_t i30;
  i30 = ((nvfuser_index_t)blockIdx.x) % 2LL;
  nvfuser_index_t i31;
  i31 = (128LL * T0.logical_size[1LL]) * i30;
  nvfuser_index_t i32;
  i32 = (256LL * T0.logical_size[1LL]) * ((nvfuser_index_t)blockIdx.y);
  __half* ptr33;
  ptr33 = (((((T0.data + i26) + i27) + i28) + i29) + i31) + i32;
  nvfuser_index_t i34;
  i34 = 16LL * T0.logical_size[1LL];
  __half* T11 = reinterpret_cast<__half*>(array + smem_offset + 0LL);
  unsigned i35;
  i35 = toSmem(T11);
  unsigned i36;
  i36 = ((((i35 + (64LL * i24)) + (64LL * (i25 / 4LL))) + (16LL * (((i24 / 2LL) % 4LL) ^ (i25 % 4LL)))) + i13) + i14;
  nvfuser_index_t i37;
  i37 = i2 / 2LL;
  unsigned i38;
  i38 = (i35 + (512LL * (i2 % 2LL))) + (64LL * i3);
  nvfuser_index_t i39;
  i39 = (i3 / 2LL) % 4LL;
  nvfuser_index_t i40;
  i40 = 4096LL * ((nvfuser_index_t)threadIdx.z);
  unsigned i41;
  i41 = ((i38 + (64LL * (i37 / 4LL))) + (16LL * (i39 ^ (i37 % 4LL)))) + i40;
  nvfuser_index_t i42;
  i42 = ((nvfuser_index_t)threadIdx.x) / 16LL;
  nvfuser_index_t i43;
  i43 = ((nvfuser_index_t)threadIdx.x) % 16LL;
  nvfuser_index_t i44;
  i44 = 128LL * i43;
  nvfuser_index_t i45;
  i45 = 8LL * (i42 % 2LL);
  nvfuser_index_t i46;
  i46 = 128LL * ((nvfuser_index_t)threadIdx.y);
  unsigned i47;
  i47 = ((i21 + i44) + i45) + i46;
  nvfuser_index_t i48;
  i48 = i42 / 2LL;
  nvfuser_index_t i49;
  i49 = i43 % 8LL;
  unsigned i50;
  i50 = ((i18 + i44) + i45) + i46;
  nvfuser_index_t i51;
  i51 = 32LL * T1.logical_size[1LL];
  __half* ptr52;
  ptr52 = (((((T2.data + i51) + i4) + i5) + i6) + i7) + i8;
  nvfuser_index_t i53;
  i53 = 32LL * T1.logical_size[1LL];
  __half* ptr54;
  ptr54 = (((((T1.data + i51) + i4) + i5) + i6) + i7) + i8;
  __half* ptr55;
  ptr55 = ((((((32LL + T0.data) + i26) + i27) + i28) + i29) + i31) + i32;
  nvfuser_index_t i56;
  i56 = 2LL + i37;
  unsigned i57;
  i57 = ((i38 + (64LL * (i56 / 4LL))) + (16LL * (i39 ^ (i56 % 4LL)))) + i40;
  unsigned i58;
  i58 = (((2048LL + i21) + i44) + i45) + i46;
  unsigned i59;
  i59 = (((2048LL + i18) + i44) + i45) + i46;
  nvfuser_index_t i60;
  i60 = 16LL * T1.logical_size[1LL];
  nvfuser_index_t i61;
  i61 = 2LL * i25;
  nvfuser_index_t i62;
  i62 = 64LL * ((nvfuser_index_t)threadIdx.y);
  nvfuser_index_t i63;
  i63 = i24 / 8LL;
  nvfuser_index_t i64;
  i64 = i24 % 8LL;
  nvfuser_index_t i65;
  i65 = ((((((i61 + (i60 * i63)) + (T1.logical_size[1LL] * i64)) + i62) + ((64LL * T1.logical_size[1LL]) * ((nvfuser_index_t)threadIdx.z))) + i8) + ((128LL * T1.logical_size[1LL]) * i30)) + ((256LL * T1.logical_size[1LL]) * ((nvfuser_index_t)blockIdx.y));
  nvfuser_index_t i66;
  i66 = 8LL * T1.logical_size[1LL];
  bool b67;
  b67 = ((7LL + (8LL * i3)) + i8) < T1.logical_size[1LL];
  nvfuser_index_t i68;
  i68 = (((7LL - (64LL * T0.logical_size[1LL])) + (8LL * ((nvfuser_index_t)threadIdx.x))) + (256LL * ((nvfuser_index_t)threadIdx.y))) + (256LL * ((nvfuser_index_t)threadIdx.z));
  nvfuser_index_t i69;
  i69 = 8LL * i25;
  bool b70;
  b70 = (7LL + i69) < T0.logical_size[1LL];
  nvfuser_index_t i71;
  i71 = -T0.logical_size[0LL];
  nvfuser_index_t i72;
  i72 = 128LL * i30;
  nvfuser_index_t i73;
  i73 = 256LL * ((nvfuser_index_t)blockIdx.y);
  nvfuser_index_t i74;
  i74 = ((((i71 + i24) + (8LL * ((nvfuser_index_t)threadIdx.y))) + (8LL * ((nvfuser_index_t)threadIdx.z))) + i72) + i73;
  nvfuser_index_t i75;
  i75 = (((32LL - T0.logical_size[1LL]) + i2) + i15) + i16;
  nvfuser_index_t i76;
  i76 = 39LL + i69;
  nvfuser_index_t i77;
  i77 = (((1LL - T1.logical_size[1LL]) + i61) + i62) + i8;
  nvfuser_index_t i78;
  i78 = ((((i71 + (16LL * i63)) + i64) + (64LL * ((nvfuser_index_t)threadIdx.z))) + i72) + i73;
  float T18[128LL];
  #pragma unroll
  for(nvfuser_index_t i79 = 0; i79 < 4LL; ++i79) {
    nvfuser_index_t i80;
    i80 = 32LL * i79;
    #pragma unroll
    for(nvfuser_index_t i81 = 0; i81 < 8LL; ++i81) {
      ((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i80 + (4LL * i81))]))).set(0);
    }
  }
  NVFUSER_UPDATE_MAGIC_ZERO;
  float T22[128LL];
  #pragma unroll
  for(nvfuser_index_t i82 = 0; i82 < 4LL; ++i82) {
    nvfuser_index_t i83;
    i83 = 32LL * i82;
    #pragma unroll
    for(nvfuser_index_t i84 = 0; i84 < 8LL; ++i84) {
      ((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i83 + (4LL * i84))]))).set(0);
    }
  }
  NVFUSER_UPDATE_MAGIC_ZERO;
  #pragma unroll
  for(nvfuser_index_t i85 = 0; i85 < 4LL; ++i85) {
    nvfuser_index_t i86;
    i86 = i85 + nvfuser_zero;
    bool b87;
    b87 = b67 && (i68 < (-(512LL * i86)));
    asm volatile(
      "{\n"
      "  .reg .pred p0; \n"
      "  setp.ne.b32 p0, %3, 0;\n"
      "  cp.async.cg.shared.global [%0], [%1], %2, p0;\n"
      "}\n"
      :
      :"r"((uint32_t)((i19 + (1024LL * i85)))),
       "l"((ptr9 + (i10 * i86))),
       "n"(16LL),
       "r"((uint32_t)((!b87)))
    );
  }
  NVFUSER_UPDATE_MAGIC_ZERO;
  #pragma unroll
  for(nvfuser_index_t i88 = 0; i88 < 4LL; ++i88) {
    nvfuser_index_t i89;
    i89 = i88 + nvfuser_zero;
    bool b90;
    b90 = b67 && (i68 < (-(512LL * i89)));
    asm volatile(
      "{\n"
      "  .reg .pred p0; \n"
      "  setp.ne.b32 p0, %3, 0;\n"
      "  cp.async.cg.shared.global [%0], [%1], %2, p0;\n"
      "}\n"
      :
      :"r"((uint32_t)((i22 + (1024LL * i88)))),
       "l"((ptr20 + (i10 * i89))),
       "n"(16LL),
       "r"((uint32_t)((!b90)))
    );
  }
  NVFUSER_UPDATE_MAGIC_ZERO;
  #pragma unroll
  for(nvfuser_index_t i91 = 0; i91 < 8LL; ++i91) {
    nvfuser_index_t i92;
    i92 = i91 + nvfuser_zero;
    bool b93;
    b93 = b70 && (i74 < (-(16LL * i92)));
    asm volatile(
      "{\n"
      "  .reg .pred p0; \n"
      "  setp.ne.b32 p0, %3, 0;\n"
      "  cp.async.cg.shared.global [%0], [%1], %2, p0;\n"
      "}\n"
      :
      :"r"((uint32_t)((i36 + (1024LL * i91)))),
       "l"((ptr33 + (i34 * i92))),
       "n"(16LL),
       "r"((uint32_t)((!b93)))
    );
  }
  NVFUSER_UPDATE_MAGIC_ZERO;
  asm volatile("cp.async.commit_group;\n");
  asm volatile("cp.async.wait_all;\n");
  __syncthreads();
  Array<__half, 64LL, 8> T23;
  Array<__half, 64LL, 4> T15;
  Array<__half, 64LL, 4> T19;
  #pragma unroll
  for(nvfuser_index_t i94 = 0; i94 < 4LL; ++i94) {
    asm volatile(
      "ldmatrix.sync.aligned.x4.m8n8.shared.b16 {%0, %1, %2, %3}, [%4];\n"
      :"=r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T23[(8LL * i94)]))[0]),
       "=r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T23[(8LL * i94)]))[1]),
       "=r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T23[(8LL * i94)]))[2]),
       "=r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T23[(8LL * i94)]))[3])
      :"r"((uint32_t)((i41 + (1024LL * i94))))
    );
  }
  NVFUSER_UPDATE_MAGIC_ZERO;
  #pragma unroll
  for(nvfuser_index_t i95 = 0; i95 < 8LL; ++i95) {
    nvfuser_index_t i96;
    i96 = i48 + i95;
    asm volatile(
      "ldmatrix.sync.aligned.x2.trans.m8n8.shared.b16 {%0, %1}, [%2];\n"
      :"=r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T15[(4LL * i95)]))[0]),
       "=r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T15[(4LL * i95)]))[1])
      :"r"((uint32_t)(((i47 + (128LL * (i96 / 8LL))) + (16LL * (i49 ^ (i96 % 8LL))))))
    );
  }
  NVFUSER_UPDATE_MAGIC_ZERO;
  #pragma unroll
  for(nvfuser_index_t i97 = 0; i97 < 8LL; ++i97) {
    nvfuser_index_t i98;
    i98 = i48 + i97;
    asm volatile(
      "ldmatrix.sync.aligned.x2.trans.m8n8.shared.b16 {%0, %1}, [%2];\n"
      :"=r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T19[(4LL * i97)]))[0]),
       "=r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T19[(4LL * i97)]))[1])
      :"r"((uint32_t)(((i50 + (128LL * (i98 / 8LL))) + (16LL * (i49 ^ (i98 % 8LL))))))
    );
  }
  NVFUSER_UPDATE_MAGIC_ZERO;
  #pragma unroll 1
  for(nvfuser_index_t i99 = 0; i99 < i0; ++i99) {
    nvfuser_index_t i100;
    i100 = i53 * i99;
    __half* ptr101;
    ptr101 = ptr52 + i100;
    nvfuser_index_t i102;
    i102 = (1LL + i99) % 2LL;
    nvfuser_index_t i103;
    i103 = 4096LL * i102;
    unsigned i104;
    i104 = i19 + i103;
    __half* ptr105;
    ptr105 = ptr54 + i100;
    unsigned i106;
    i106 = i22 + i103;
    nvfuser_index_t i107;
    i107 = 32LL * i99;
    __half* ptr108;
    ptr108 = ptr55 + i107;
    nvfuser_index_t i109;
    i109 = 8192LL * i102;
    unsigned i110;
    i110 = i36 + i109;
    nvfuser_index_t i111;
    i111 = i99 % 2LL;
    unsigned i112;
    i112 = i57 + (8192LL * i111);
    nvfuser_index_t i113;
    i113 = 4096LL * i111;
    unsigned i114;
    i114 = i58 + i113;
    unsigned i115;
    i115 = i59 + i113;
    unsigned i116;
    i116 = i41 + i109;
    unsigned i117;
    i117 = i47 + i103;
    unsigned i118;
    i118 = i50 + i103;
    nvfuser_index_t i119;
    i119 = i75 + i107;
    bool b120;
    b120 = (i76 + i107) < T0.logical_size[1LL];
    #pragma unroll
    for(nvfuser_index_t i85 = 0; i85 < 4LL; ++i85) {
      nvfuser_index_t i121;
      i121 = i85 + nvfuser_zero;
      bool b122;
      b122 = b67 && (i119 < (-(8LL * i121)));
      asm volatile(
        "{\n"
        "  .reg .pred p0; \n"
        "  setp.ne.b32 p0, %3, 0;\n"
        "  cp.async.cg.shared.global [%0], [%1], %2, p0;\n"
        "}\n"
        :
        :"r"((uint32_t)((i104 + (1024LL * i85)))),
         "l"((ptr101 + (i10 * i121))),
         "n"(16LL),
         "r"((uint32_t)((!b122)))
      );
    }
    NVFUSER_UPDATE_MAGIC_ZERO;
    #pragma unroll
    for(nvfuser_index_t i88 = 0; i88 < 4LL; ++i88) {
      nvfuser_index_t i123;
      i123 = i88 + nvfuser_zero;
      bool b124;
      b124 = b67 && (i119 < (-(8LL * i123)));
      asm volatile(
        "{\n"
        "  .reg .pred p0; \n"
        "  setp.ne.b32 p0, %3, 0;\n"
        "  cp.async.cg.shared.global [%0], [%1], %2, p0;\n"
        "}\n"
        :
        :"r"((uint32_t)((i106 + (1024LL * i88)))),
         "l"((ptr105 + (i10 * i123))),
         "n"(16LL),
         "r"((uint32_t)((!b124)))
      );
    }
    NVFUSER_UPDATE_MAGIC_ZERO;
    #pragma unroll
    for(nvfuser_index_t i91 = 0; i91 < 8LL; ++i91) {
      nvfuser_index_t i125;
      i125 = i91 + nvfuser_zero;
      bool b126;
      b126 = b120 && (i74 < (-(16LL * i125)));
      asm volatile(
        "{\n"
        "  .reg .pred p0; \n"
        "  setp.ne.b32 p0, %3, 0;\n"
        "  cp.async.cg.shared.global [%0], [%1], %2, p0;\n"
        "}\n"
        :
        :"r"((uint32_t)((i110 + (1024LL * i91)))),
         "l"((ptr108 + (i34 * i125))),
         "n"(16LL),
         "r"((uint32_t)((!b126)))
      );
    }
    NVFUSER_UPDATE_MAGIC_ZERO;
    asm volatile("cp.async.commit_group;\n");
    #pragma unroll
    for(nvfuser_index_t i94 = 0; i94 < 4LL; ++i94) {
      asm volatile(
        "ldmatrix.sync.aligned.x4.m8n8.shared.b16 {%0, %1, %2, %3}, [%4];\n"
        :"=r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T23[(32LL + (8LL * i94))]))[0]),
         "=r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T23[(32LL + (8LL * i94))]))[1]),
         "=r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T23[(32LL + (8LL * i94))]))[2]),
         "=r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T23[(32LL + (8LL * i94))]))[3])
        :"r"((uint32_t)((i112 + (1024LL * i94))))
      );
    }
    __half T16[32LL];
    #pragma unroll
    for(nvfuser_index_t i127 = 0; i127 < 4LL; ++i127) {
      nvfuser_index_t i128;
      i128 = 8LL * i127;
      #pragma unroll
      for(nvfuser_index_t i129 = 0; i129 < 2LL; ++i129) {
        nvfuser_index_t i130;
        i130 = i128 + (4LL * i129);
        #pragma unroll
        for(nvfuser_index_t i131 = 0; i131 < 2LL; ++i131) {
          nvfuser_index_t i132;
          i132 = i130 + (2LL * i131);
          #pragma unroll
          for(nvfuser_index_t i133 = 0; i133 < 2LL; ++i133) {
            T16[(i132 + i133)] = 0.000000000e+00f;
          }
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i127 = 0; i127 < 4LL; ++i127) {
      nvfuser_index_t i134;
      i134 = 8LL * i127;
      #pragma unroll
      for(nvfuser_index_t i129 = 0; i129 < 2LL; ++i129) {
        nvfuser_index_t i135;
        i135 = i134 + (4LL * i129);
        #pragma unroll
        for(nvfuser_index_t i131 = 0; i131 < 2LL; ++i131) {
          nvfuser_index_t i136;
          i136 = i135 + (2LL * i131);
          #pragma unroll
          for(nvfuser_index_t i133 = 0; i133 < 2LL; ++i133) {
            nvfuser_index_t i137;
            i137 = i136 + i133;
            T16[i137]
               = T23[i137];
          }
        }
      }
    }
    __half T20[32LL];
    #pragma unroll
    for(nvfuser_index_t i138 = 0; i138 < 4LL; ++i138) {
      nvfuser_index_t i139;
      i139 = 8LL * i138;
      #pragma unroll
      for(nvfuser_index_t i140 = 0; i140 < 2LL; ++i140) {
        nvfuser_index_t i141;
        i141 = i139 + (4LL * i140);
        #pragma unroll
        for(nvfuser_index_t i142 = 0; i142 < 2LL; ++i142) {
          nvfuser_index_t i143;
          i143 = i141 + (2LL * i142);
          #pragma unroll
          for(nvfuser_index_t i144 = 0; i144 < 2LL; ++i144) {
            T20[(i143 + i144)] = 0.000000000e+00f;
          }
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i138 = 0; i138 < 4LL; ++i138) {
      nvfuser_index_t i145;
      i145 = 8LL * i138;
      #pragma unroll
      for(nvfuser_index_t i140 = 0; i140 < 2LL; ++i140) {
        nvfuser_index_t i146;
        i146 = i145 + (4LL * i140);
        #pragma unroll
        for(nvfuser_index_t i142 = 0; i142 < 2LL; ++i142) {
          nvfuser_index_t i147;
          i147 = i146 + (2LL * i142);
          #pragma unroll
          for(nvfuser_index_t i144 = 0; i144 < 2LL; ++i144) {
            nvfuser_index_t i148;
            i148 = i147 + i144;
            T20[i148]
               = T23[i148];
          }
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i95 = 0; i95 < 8LL; ++i95) {
      nvfuser_index_t i149;
      i149 = i48 + i95;
      asm volatile(
        "ldmatrix.sync.aligned.x2.trans.m8n8.shared.b16 {%0, %1}, [%2];\n"
        :"=r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T15[(32LL + (4LL * i95))]))[0]),
         "=r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T15[(32LL + (4LL * i95))]))[1])
        :"r"((uint32_t)(((i114 + (128LL * (i149 / 8LL))) + (16LL * (i49 ^ (i149 % 8LL))))))
      );
    }
    __half T17[32LL];
    #pragma unroll
    for(nvfuser_index_t i150 = 0; i150 < 8LL; ++i150) {
      nvfuser_index_t i151;
      i151 = 4LL * i150;
      #pragma unroll
      for(nvfuser_index_t i152 = 0; i152 < 2LL; ++i152) {
        nvfuser_index_t i153;
        i153 = i151 + (2LL * i152);
        #pragma unroll
        for(nvfuser_index_t i154 = 0; i154 < 2LL; ++i154) {
          T17[(i153 + i154)] = 0.000000000e+00f;
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i150 = 0; i150 < 8LL; ++i150) {
      nvfuser_index_t i155;
      i155 = 4LL * i150;
      #pragma unroll
      for(nvfuser_index_t i152 = 0; i152 < 2LL; ++i152) {
        nvfuser_index_t i156;
        i156 = i155 + (2LL * i152);
        #pragma unroll
        for(nvfuser_index_t i154 = 0; i154 < 2LL; ++i154) {
          nvfuser_index_t i157;
          i157 = i156 + i154;
          T17[i157]
             = T15[i157];
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i79 = 0; i79 < 4LL; ++i79) {
      nvfuser_index_t i158;
      i158 = 8LL * i79;
      nvfuser_index_t i159;
      i159 = 32LL * i79;
      #pragma unroll
      for(nvfuser_index_t i81 = 0; i81 < 8LL; ++i81) {
        nvfuser_index_t i160;
        i160 = 4LL * i81;
        asm(
          "mma.sync.aligned.m16n8k16.row.col.f32.f16.f16.f32 {%0, %1, %2, %3}, {%4, %5, %6, %7}, {%8, %9}, {%10, %11, %12, %13};\n"
          :"=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i159 + i160)]))[0]),
           "=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i159 + i160)]))[1]),
           "=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i159 + i160)]))[2]),
           "=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i159 + i160)]))[3])
          :"r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T16[i158]))[0]),
           "r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T16[i158]))[1]),
           "r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T16[i158]))[2]),
           "r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T16[i158]))[3]),
           "r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T17[i160]))[0]),
           "r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T17[i160]))[1]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i159 + i160)]))[0]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i159 + i160)]))[1]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i159 + i160)]))[2]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i159 + i160)]))[3])
        );
      }
    }
    #pragma unroll
    for(nvfuser_index_t i97 = 0; i97 < 8LL; ++i97) {
      nvfuser_index_t i161;
      i161 = i48 + i97;
      asm volatile(
        "ldmatrix.sync.aligned.x2.trans.m8n8.shared.b16 {%0, %1}, [%2];\n"
        :"=r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T19[(32LL + (4LL * i97))]))[0]),
         "=r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T19[(32LL + (4LL * i97))]))[1])
        :"r"((uint32_t)(((i115 + (128LL * (i161 / 8LL))) + (16LL * (i49 ^ (i161 % 8LL))))))
      );
    }
    // Alias Allocation - register
    auto& T21 = T16;
    #pragma unroll
    for(nvfuser_index_t i162 = 0; i162 < 8LL; ++i162) {
      nvfuser_index_t i163;
      i163 = 4LL * i162;
      #pragma unroll
      for(nvfuser_index_t i164 = 0; i164 < 2LL; ++i164) {
        nvfuser_index_t i165;
        i165 = i163 + (2LL * i164);
        #pragma unroll
        for(nvfuser_index_t i166 = 0; i166 < 2LL; ++i166) {
          T21[(i165 + i166)] = 0.000000000e+00f;
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i162 = 0; i162 < 8LL; ++i162) {
      nvfuser_index_t i167;
      i167 = 4LL * i162;
      #pragma unroll
      for(nvfuser_index_t i164 = 0; i164 < 2LL; ++i164) {
        nvfuser_index_t i168;
        i168 = i167 + (2LL * i164);
        #pragma unroll
        for(nvfuser_index_t i166 = 0; i166 < 2LL; ++i166) {
          nvfuser_index_t i169;
          i169 = i168 + i166;
          T21[i169]
             = T19[i169];
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i82 = 0; i82 < 4LL; ++i82) {
      nvfuser_index_t i170;
      i170 = 8LL * i82;
      nvfuser_index_t i171;
      i171 = 32LL * i82;
      #pragma unroll
      for(nvfuser_index_t i84 = 0; i84 < 8LL; ++i84) {
        nvfuser_index_t i172;
        i172 = 4LL * i84;
        asm(
          "mma.sync.aligned.m16n8k16.row.col.f32.f16.f16.f32 {%0, %1, %2, %3}, {%4, %5, %6, %7}, {%8, %9}, {%10, %11, %12, %13};\n"
          :"=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i171 + i172)]))[0]),
           "=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i171 + i172)]))[1]),
           "=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i171 + i172)]))[2]),
           "=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i171 + i172)]))[3])
          :"r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T20[i170]))[0]),
           "r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T20[i170]))[1]),
           "r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T20[i170]))[2]),
           "r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T20[i170]))[3]),
           "r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T21[i172]))[0]),
           "r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T21[i172]))[1]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i171 + i172)]))[0]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i171 + i172)]))[1]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i171 + i172)]))[2]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i171 + i172)]))[3])
        );
      }
    }
    NVFUSER_UPDATE_MAGIC_ZERO;
    #pragma unroll
    for(nvfuser_index_t i127 = 0; i127 < 4LL; ++i127) {
      nvfuser_index_t i173;
      i173 = 8LL * i127;
      #pragma unroll
      for(nvfuser_index_t i129 = 0; i129 < 2LL; ++i129) {
        nvfuser_index_t i174;
        i174 = i173 + (4LL * i129);
        #pragma unroll
        for(nvfuser_index_t i131 = 0; i131 < 2LL; ++i131) {
          nvfuser_index_t i175;
          i175 = i174 + (2LL * i131);
          #pragma unroll
          for(nvfuser_index_t i133 = 0; i133 < 2LL; ++i133) {
            T16[(i175 + i133)] = 0.000000000e+00f;
          }
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i127 = 0; i127 < 4LL; ++i127) {
      nvfuser_index_t i176;
      i176 = 8LL * i127;
      nvfuser_index_t i177;
      i177 = 32LL + i176;
      #pragma unroll
      for(nvfuser_index_t i129 = 0; i129 < 2LL; ++i129) {
        nvfuser_index_t i178;
        i178 = 4LL * i129;
        nvfuser_index_t i179;
        i179 = i176 + i178;
        nvfuser_index_t i180;
        i180 = i177 + i178;
        #pragma unroll
        for(nvfuser_index_t i131 = 0; i131 < 2LL; ++i131) {
          nvfuser_index_t i181;
          i181 = 2LL * i131;
          nvfuser_index_t i182;
          i182 = i179 + i181;
          nvfuser_index_t i183;
          i183 = i180 + i181;
          #pragma unroll
          for(nvfuser_index_t i133 = 0; i133 < 2LL; ++i133) {
            T16[(i182 + i133)]
               = T23[(i183 + i133)];
          }
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i138 = 0; i138 < 4LL; ++i138) {
      nvfuser_index_t i184;
      i184 = 8LL * i138;
      #pragma unroll
      for(nvfuser_index_t i140 = 0; i140 < 2LL; ++i140) {
        nvfuser_index_t i185;
        i185 = i184 + (4LL * i140);
        #pragma unroll
        for(nvfuser_index_t i142 = 0; i142 < 2LL; ++i142) {
          nvfuser_index_t i186;
          i186 = i185 + (2LL * i142);
          #pragma unroll
          for(nvfuser_index_t i144 = 0; i144 < 2LL; ++i144) {
            T20[(i186 + i144)] = 0.000000000e+00f;
          }
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i138 = 0; i138 < 4LL; ++i138) {
      nvfuser_index_t i187;
      i187 = 8LL * i138;
      nvfuser_index_t i188;
      i188 = 32LL + i187;
      #pragma unroll
      for(nvfuser_index_t i140 = 0; i140 < 2LL; ++i140) {
        nvfuser_index_t i189;
        i189 = 4LL * i140;
        nvfuser_index_t i190;
        i190 = i187 + i189;
        nvfuser_index_t i191;
        i191 = i188 + i189;
        #pragma unroll
        for(nvfuser_index_t i142 = 0; i142 < 2LL; ++i142) {
          nvfuser_index_t i192;
          i192 = 2LL * i142;
          nvfuser_index_t i193;
          i193 = i190 + i192;
          nvfuser_index_t i194;
          i194 = i191 + i192;
          #pragma unroll
          for(nvfuser_index_t i144 = 0; i144 < 2LL; ++i144) {
            T20[(i193 + i144)]
               = T23[(i194 + i144)];
          }
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i150 = 0; i150 < 8LL; ++i150) {
      nvfuser_index_t i195;
      i195 = 4LL * i150;
      #pragma unroll
      for(nvfuser_index_t i152 = 0; i152 < 2LL; ++i152) {
        nvfuser_index_t i196;
        i196 = i195 + (2LL * i152);
        #pragma unroll
        for(nvfuser_index_t i154 = 0; i154 < 2LL; ++i154) {
          T17[(i196 + i154)] = 0.000000000e+00f;
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i150 = 0; i150 < 8LL; ++i150) {
      nvfuser_index_t i197;
      i197 = 4LL * i150;
      nvfuser_index_t i198;
      i198 = 32LL + i197;
      #pragma unroll
      for(nvfuser_index_t i152 = 0; i152 < 2LL; ++i152) {
        nvfuser_index_t i199;
        i199 = 2LL * i152;
        nvfuser_index_t i200;
        i200 = i197 + i199;
        nvfuser_index_t i201;
        i201 = i198 + i199;
        #pragma unroll
        for(nvfuser_index_t i154 = 0; i154 < 2LL; ++i154) {
          T17[(i200 + i154)]
             = T15[(i201 + i154)];
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i79 = 0; i79 < 4LL; ++i79) {
      nvfuser_index_t i202;
      i202 = 8LL * i79;
      nvfuser_index_t i203;
      i203 = 32LL * i79;
      #pragma unroll
      for(nvfuser_index_t i81 = 0; i81 < 8LL; ++i81) {
        nvfuser_index_t i204;
        i204 = 4LL * i81;
        asm(
          "mma.sync.aligned.m16n8k16.row.col.f32.f16.f16.f32 {%0, %1, %2, %3}, {%4, %5, %6, %7}, {%8, %9}, {%10, %11, %12, %13};\n"
          :"=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i203 + i204)]))[0]),
           "=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i203 + i204)]))[1]),
           "=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i203 + i204)]))[2]),
           "=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i203 + i204)]))[3])
          :"r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T16[i202]))[0]),
           "r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T16[i202]))[1]),
           "r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T16[i202]))[2]),
           "r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T16[i202]))[3]),
           "r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T17[i204]))[0]),
           "r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T17[i204]))[1]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i203 + i204)]))[0]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i203 + i204)]))[1]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i203 + i204)]))[2]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T18[(i203 + i204)]))[3])
        );
      }
    }
    #pragma unroll
    for(nvfuser_index_t i162 = 0; i162 < 8LL; ++i162) {
      nvfuser_index_t i205;
      i205 = 4LL * i162;
      #pragma unroll
      for(nvfuser_index_t i164 = 0; i164 < 2LL; ++i164) {
        nvfuser_index_t i206;
        i206 = i205 + (2LL * i164);
        #pragma unroll
        for(nvfuser_index_t i166 = 0; i166 < 2LL; ++i166) {
          T21[(i206 + i166)] = 0.000000000e+00f;
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i162 = 0; i162 < 8LL; ++i162) {
      nvfuser_index_t i207;
      i207 = 4LL * i162;
      nvfuser_index_t i208;
      i208 = 32LL + i207;
      #pragma unroll
      for(nvfuser_index_t i164 = 0; i164 < 2LL; ++i164) {
        nvfuser_index_t i209;
        i209 = 2LL * i164;
        nvfuser_index_t i210;
        i210 = i207 + i209;
        nvfuser_index_t i211;
        i211 = i208 + i209;
        #pragma unroll
        for(nvfuser_index_t i166 = 0; i166 < 2LL; ++i166) {
          T21[(i210 + i166)]
             = T19[(i211 + i166)];
        }
      }
    }
    #pragma unroll
    for(nvfuser_index_t i82 = 0; i82 < 4LL; ++i82) {
      nvfuser_index_t i212;
      i212 = 8LL * i82;
      nvfuser_index_t i213;
      i213 = 32LL * i82;
      #pragma unroll
      for(nvfuser_index_t i84 = 0; i84 < 8LL; ++i84) {
        nvfuser_index_t i214;
        i214 = 4LL * i84;
        asm(
          "mma.sync.aligned.m16n8k16.row.col.f32.f16.f16.f32 {%0, %1, %2, %3}, {%4, %5, %6, %7}, {%8, %9}, {%10, %11, %12, %13};\n"
          :"=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i213 + i214)]))[0]),
           "=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i213 + i214)]))[1]),
           "=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i213 + i214)]))[2]),
           "=f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i213 + i214)]))[3])
          :"r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T20[i212]))[0]),
           "r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T20[i212]))[1]),
           "r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T20[i212]))[2]),
           "r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T20[i212]))[3]),
           "r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T21[i214]))[0]),
           "r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T21[i214]))[1]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i213 + i214)]))[0]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i213 + i214)]))[1]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i213 + i214)]))[2]),
           "f"((*reinterpret_cast<Array<float, 4, 1>*>(&T22[(i213 + i214)]))[3])
        );
      }
    }
    NVFUSER_UPDATE_MAGIC_ZERO;
    asm volatile("cp.async.wait_all;\n");
    __syncthreads();
    #pragma unroll
    for(nvfuser_index_t i94 = 0; i94 < 4LL; ++i94) {
      asm volatile(
        "ldmatrix.sync.aligned.x4.m8n8.shared.b16 {%0, %1, %2, %3}, [%4];\n"
        :"=r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T23[(8LL * i94)]))[0]),
         "=r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T23[(8LL * i94)]))[1]),
         "=r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T23[(8LL * i94)]))[2]),
         "=r"((*reinterpret_cast<Array<uint32_t, 4, 1>*>(&T23[(8LL * i94)]))[3])
        :"r"((uint32_t)((i116 + (1024LL * i94))))
      );
    }
    NVFUSER_UPDATE_MAGIC_ZERO;
    #pragma unroll
    for(nvfuser_index_t i95 = 0; i95 < 8LL; ++i95) {
      nvfuser_index_t i215;
      i215 = i48 + i95;
      asm volatile(
        "ldmatrix.sync.aligned.x2.trans.m8n8.shared.b16 {%0, %1}, [%2];\n"
        :"=r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T15[(4LL * i95)]))[0]),
         "=r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T15[(4LL * i95)]))[1])
        :"r"((uint32_t)(((i117 + (128LL * (i215 / 8LL))) + (16LL * (i49 ^ (i215 % 8LL))))))
      );
    }
    NVFUSER_UPDATE_MAGIC_ZERO;
    #pragma unroll
    for(nvfuser_index_t i97 = 0; i97 < 8LL; ++i97) {
      nvfuser_index_t i216;
      i216 = i48 + i97;
      asm volatile(
        "ldmatrix.sync.aligned.x2.trans.m8n8.shared.b16 {%0, %1}, [%2];\n"
        :"=r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T19[(4LL * i97)]))[0]),
         "=r"((*reinterpret_cast<Array<uint32_t, 2, 1>*>(&T19[(4LL * i97)]))[1])
        :"r"((uint32_t)(((i118 + (128LL * (i216 / 8LL))) + (16LL * (i49 ^ (i216 % 8LL))))))
      );
    }
    NVFUSER_UPDATE_MAGIC_ZERO;
  }
  asm volatile("cp.async.wait_all;\n");
  #pragma unroll
  for(nvfuser_index_t i217 = 0; i217 < 4LL; ++i217) {
    nvfuser_index_t i218;
    i218 = 32LL * i217;
    nvfuser_index_t i219;
    i219 = i65 + (i60 * i217);
    nvfuser_index_t i220;
    i220 = -(16LL * i217);
    #pragma unroll
    for(nvfuser_index_t i221 = 0; i221 < 8LL; ++i221) {
      nvfuser_index_t i222;
      i222 = i218 + (4LL * i221);
      nvfuser_index_t i223;
      i223 = i219 + (8LL * i221);
      bool b224;
      b224 = i77 < (-(8LL * (i221 + nvfuser_zero)));
      #pragma unroll
      for(nvfuser_index_t i225 = 0; i225 < 2LL; ++i225) {
        nvfuser_index_t i226;
        i226 = i222 + (2LL * i225);
        nvfuser_index_t i227;
        i227 = i225 + nvfuser_zero;
        Array<float, 2LL, 2> T14;
        #pragma unroll
        for(nvfuser_index_t i228 = 0; i228 < 2LL; ++i228) {
          nvfuser_index_t i229;
          i229 = i226 + i228;
          float T6[1LL];
          T6[0LL]
             = sigmoid(T18[i229]);
          float T8[1LL];
          T8[0LL]
            = T6[0LL]
            * T18[i229];
          T14[i228]
            = T8[0LL]
            * T22[i229];
        }
        if ((b224 && (i78 < (i220 - (8LL * i227))))) {
          loadLocalToGlobal<float, /*vec_size=*/2, /*is_volatile=*/false>( &T10[(i223 + (i66 * i227))], &T14[0LL]);
        }
      }
    }
  }
  NVFUSER_UPDATE_MAGIC_ZERO;
}
```